Code/GraphMol/FileParsers/MolSGroupParsing.cpp

```diff
@@ -642,14 +642,15 @@ void ParseSGroupV2000SCDSEDLine(IDX_TO_SGROUP_MAP &sGroupMap,
     lastDataSGroup = 0;
   }
 
-  // this group must already have seen an SDT line
+  // have we already seen an SDT line?
   if (!sgroup->hasProp("FIELDNAME")) {
```

Comment:
Would it make sense to keep the old behavior if we request script parsing?

Reply:
I thought about that, but given that strict parsing is the default and it's not 100% clear that this is wrong, I opted to just issue the warning.

Reply:
Oh, I think it is wrong:

![image](https://user-images.githubusercontent.com/7498185/110847080-7a098f80-827a-11eb-941e-d39a94fcc7f8.png)

So, the line *must* be present if there are data lines, and the SDT line itself shouldn't be blank either, since at least the field type should indicate a type of data (either N or T). And that assuming that we allow the field name to be blank, which is not clear:

![image](https://user-images.githubusercontent.com/7498185/110847219-a58c7a00-827a-11eb-902c-1490c72fa7ad.png)

Reply:
sorry, you're right abuot the missing SDT line. 

The bit about the line not being full of blanks is a different question.
I could interpret `30 character field name - no blanks, commas, or hyphens for MACCS-II` to mean that it's ok to have blanks, commas, or hyphens if we aren't using MACCS-II.

We definitely could continue to throw an exception if strictParsing is true and people who want to parse this thing permissively can update their code. If you think that's a better idea I can add the exception again.

Reply:
> The bit about the line not being full of blanks is a different question.
> I could interpret `30 character field name - no blanks, commas, or hyphens for MACCS-II` to mean that it's ok to have blanks, commas, or hyphens if we aren't using MACCS-II.

Yeah, it's like one of those optical illusions, every time I look at that paragraph, I change my mind about what it means: either no blanks nor commas at all, and also no hyphens for MACCS-II, or all of those are allowed except if the type is MACCS-II, and then none is allowed.

In any case, the next field in the line, the field type, seems to only accept "N" or "T" as values, so it definitely shouldn't be blank.

> We definitely could continue to throw an exception if strictParsing is true and people who want to parse this thing permissively can update their code. If you think that's a better idea I can add the exception again.

I think we can leave it as it is: we are not doing anything with that data, and anyone using it will most probably validate it on their side before using it.